batch/batch.go

```diff
@@ -119,15 +119,18 @@ func (b *Builder) Build(now time.Time) (*Batch, error) {
 
 var (
 	CorruptedBatchError = errors.New("batch crc does not match bytes")
+	UnsupportedProtocolError = errors.New(" protocol version with magic value 2 is only supported")
 	crc32c              = crc32.MakeTable(crc32.Castagnoli)
 )
-
 // Unmarshal the batch. On error batch is nil. If there is an error, it is most
 // likely because the crc failed. In that case there is no way to tell how many
 // records there were in the batch (and to adjust offsets accordingly).
 func Unmarshal(b []byte) (*Batch, error) {
 	buf := bytes.NewBuffer(b)
 	batch := &Batch{}
+	if b[unsafe.Offsetof(batch.Magic)] != 2 {
```

Comment:
i don't really understand why we need to retrieve the offset of the field from the struct using `unsafe`. should we keep it simple and have a constant containing the index of the magic byte?

Reply:
I feel that too